### PR TITLE
chore(federation): Update schema printers

### LINKED ISSUE #635: fix(query-planner-js): `buildComposedSchema` doesn't preserve `@deprecated` usages on input values
Reported by @sachindshinde - specifically calling `printSchema(buildComposedSchema(parse(supergraphSdl)))` loses `@deprecated` usages on input values.

### LINKED ISSUE #994: `printSupergraphSdl` and `printFederatedSchema` should support `@specifiedBy`
Both `printSupergraphSdl` and `printFederatedSchema` are forks of `printSchema` from `graphql-js`, and have not yet been updated to output `@specifiedBy` for scalar types (see [here](https://github.com/graphql/graphql-js/blob/da685ee1edcffe46bb6902ed48318f0f1905065e/src/utilities/printSchema.ts#L295-L301) for the corresponding code in `graphql-js`).

As far as I can see, this means we don't currently support `@specifiedBy` in subgraphs when using `@apollo/federation` (it won't show up under `{ _service { sdl } }`). And even though composition will keep `@specifiedBy` on type definitions, it won't show up in the printed supergraph SDL. (This means it may actually have worked for some people before the switch to supergraph SDL, when using local composition in the gateway with a non-Node server that does support `@specifiedBy` for subgraphs.)

### PR DESCRIPTION
**Pull request by trevor-scheer**
Update our subgraph and supergraph schema printers to match the `printSchema` function from `graphql-js` as closely as possible. This introduces printing capabilities for the `@specifiedBy` directive as well as `@deprecated` on input values.

This PR depends on #1000 which resolves a few problems I turned up after addressing @martijnwalraven's sorting concern.
* The supergraph printer was doing some composition-level operations on the schema (adding directives and types). I pushed this behavior more appropriately up to composition where it belongs.
* This change resulted in supergraphs being sorted correctly (`lexicographicSortSchema` was being called, but the printer was modifying the schema after that sort).
* This change was also reflected in the subgraph printer tests which was odd. I found that the `printFederatedSchema` test was testing printing a composed graph rather than a subgraph, so I fixed the test.
* Alias and deprecate `printFederatedSchema` (which was exported as `printSchema`) to `printSubgraphSchema` to prevent naming confusion.

Fixes #994
Fixes #635

### CODE REVIEW

**Comment by trevor-scheer** [comment]
Needs changelog entry

**Review by martijnwalraven — APPROVED** [review]
Thanks for tackling this! I don't have any real feedback on these changes in particular. Some changes seem unrelated to the `@specifiedBy`/`@deprecated` additions, but I'm assuming that's because you're brought the code up to date with `graphql-js` v16.

We've talked about this before, but it's a shame we have to fork `printSchema` (and do it twice!). This is probably a good topic for discussion with @IvanGoncharov, but it would be great to investigate additions to `graphql-js` that could avoid the need for forking altogether.

File: federation-js/src/service/printFederatedSchema.ts [hunk 1]
```diff
@@ -95,19 +68,19 @@ function printFilteredSchema(
   schema: GraphQLSchema,
   directiveFilter: (type: GraphQLDirective) => boolean,
   typeFilter: (type: GraphQLNamedType) => boolean,
-  options?: Options,
 ): string {
   const directives = schema.getDirectives().filter(directiveFilter);
   const types = Object.values(schema.getTypeMap())
+    // Apollo addition - sort types by name
     .sort((type1, type2) => type1.name.localeCompare(type2.name))
```
[the remark this answers is not in the document]
Comment (trevor-scheer) [reply federation-js/src/service/printFederatedSchema.ts:75]: Good catch, and definitely agree.

After some discussion we've agreed to ensure that the schema object itself is sorted as an output of composition. The composed schema is an artifact of composition which belongs to and is created by the library. We think it makes sense to have the result of composition be deterministic regardless of the ordering of serviceList, etc.

However, the sorting of subgraphs (`printFederatedSchema`) is unnecessary and is possibly even a disservice to our users.

**Review by martijnwalraven — APPROVED** [review]
Looks good to me!

File: federation-js/src/service/printSupergraphSdl.ts [hunk 2]
```diff
@@ -504,72 +492,71 @@ function printInputValue(arg: GraphQLInputField) {
...
   return ' @deprecated';
 }
 
-function printDescription<T extends { description?: Maybe<string> }>(
-  options: Options | undefined,
-  def: T,
-  indentation = '',
-  firstInBlock = true,
+// Apollo addition: support both specifiedByUrl and specifiedByURL - these
+// happen across v15 and v16.
```
Comment (trevor-scheer) [inline federation-js/src/service/printSupergraphSdl.ts:522]: @IvanGoncharov do you intend to preserve this breaking change in v16?
https://github.com/graphql/graphql-js/issues/3156

File: federation-js/src/service/printFederatedSchema.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (trevor-scheer) [reply federation-js/src/service/printFederatedSchema.ts]: Resolved in #1000

**Pull request merged**